src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -520,13 +520,22 @@ void GcsActorManager::DestroyActor(const ActorID &actor_id) {
       // The actor was being scheduled and has now been canceled.
       RAY_CHECK(canceled_actor_id == actor_id);
     } else {
-      // The actor was pending scheduling. Remove it from the queue.
       auto pending_it = std::find_if(pending_actors_.begin(), pending_actors_.end(),
                                      [actor_id](const std::shared_ptr<GcsActor> &actor) {
                                        return actor->GetActorID() == actor_id;
                                      });
-      RAY_CHECK(pending_it != pending_actors_.end());
-      pending_actors_.erase(pending_it);
+      // The actor was pending scheduling. Remove it from the queue.
+      if (pending_it != pending_actors_.end()) {
+        pending_actors_.erase(pending_it);
+      } else {
+        // When actor creation request of this actor id is pending in raylet,
+        // it doesn't responds, and the actor should be still in leasing state.
+        // TODO(sang): Theoretically, worker leaks should not happen because
+        // gcs will publish the actor dead state to raylet after, and it will
+        // clean up leased workers.
+        // Call raylet->CancelWorkerLease explicitly if worker leaks are detected.
```

Comment:
I didn't find the code to call raylet->CancelWorkerLease explicitly if worker leaks are detected.

Reply:
Sorry for vague wording. I will change the comment. What I meant was that from my experiment (and there's a test), I confirmed worker leaks don't happen although we don't call `raylet->CancelWorkerLease` because published DEAD message will handle this. I just wanted to add a todo message as a warning.